### ISSUE TITLE
Add Warning for CIS Profile Configuration on Running RKE2 Clusters

### ISSUE THREAD

**Issue by skanakal** [rancher/dashboard#11873]
**Setup**
- Rancher version: 2.9.1
- Rancher UI Extensions:
- Browser type & version: any

**Describe the bug**

When enabling the CIS profile on an already provisioned RKE2 cluster, users must manually stop the rke2-server process on etcd nodes to prevent issues with the etcd database being memory-mapped and open for writing. The current [documentation](https://docs.rke2.io/security/hardening_guide#host-level-requirements) highlights this requirement but it’s not clearly communicated in the UI.

**To Reproduce**

- Provision a single-node RKE2 Rancher-provisioned custom cluster. 
- Once the cluster is running Set the CIS Profile to `cis` or `cis-1.23`, as appropriate for the Kubernetes Version, via Edit Config for the Rancher-provisioned cluster and run `systemctl restart rke2-server` 

**Result**
The UI does not currently provide any warning about stopping the rke2-server process, potentially leading to confusion and etcd container failed to start with permission denied errors when enabling CIS profile on a running cluster.

**Expected Result**

A prominent warning or banner should be added to the UI, alerting users that they need to stop the rke2-server service before enabling the CIS profile on an already running RKE2 cluster.

**Screenshots**

They're something similar to this:

![image](https://github.com/user-attachments/assets/27b7a561-53c7-4097-813a-51b34e29f9f1)

**Additional context**

Internal ref: SURE-8934

**Comment by momesgin** [comment]
@skanakal thank you for creating the issue, I have a few questions:

1. Should the warning be shown only for custom clusters and only when the user is in Editing mode and selects a CIS profile?
2. The warning in the screenshot you shared looks like it was manually created by you, right?
3. Do we also want to include the documentation link you mentioned in the warning description?

cc @kwwii

**Comment by skanakal** [comment]
@momesgin Thank you for looking into it.
I think this needs to be addressed from backend first considering HA clusters... I have added my response in SURE-8934.
 
> 1. Should the warning be shown only for custom clusters and only when the user is in Editing mode and selects a CIS profile?

Yes, that's correct. This is only applicable for custom clusters.

> 2. The warning in the screenshot you shared looks like it was manually created by you, right?

Yeah, I have changed the dashboard code to show an example...

> 3. Do we also want to include the documentation link you mentioned in the warning description?

Yeah that would be helpful...

**Comment by nwmac** [comment]
@momesgin to chase this one up and move this and the PR out to 2.13 if necessary

**Comment by momesgin** [comment]
@skanakal I checked the JIRA tickets again, and I think the UI shouldn't be blocked on this anymore. For the warning message I'm gonna go with this:

> _If you have already started RKE2, you will need to stop RKE2 service before enabling the Compliance profile. For more information see our [documentation](https://docs.rke2.io/security/hardening_guide#host-level-requirements)._

but let me know if it should be different.

**Comment by skanakal** [comment]
@momesgin Yep, that’s right. This is just a temporary banner until the [backend issue](https://github.com/rancher/rancher/issues/50080) is resolved. We can remove it afterward...

**Comment by richard-cox** [comment]
@momesgin can you create an issue to revert this change and mark is a blocked on https://github.com/rancher/rancher/issues/50080?

@skanakal is there a target milestone for that change?

**Comment by momesgin** [comment]
The issue for reverting this change is created
https://github.com/rancher/dashboard/issues/14677

**Comment by a-arias** [comment]
**TESTS** 
Rancher : v2.12-122ba407047ed762689d3778c1bdafb4fbb91004-head
UI: master 6e5f1be 
New message is displayed ✅ 

<img width="1333" height="753" alt="Image" src="https://github.com/user-attachments/assets/0df61088-023f-4f17-8939-ed35639f13aa" />
